### PR TITLE
refactor: upgrade from CRA to Vite

### PR DESCRIPTION
**Pull request by Mohammer5**
I've added our dhis2 linter & prettier configs (by just copying them over). I haven't applied them so I can rebase / merge any changes in master while this PR is still in review.

If you want to check out the amount of changes, just run `yarn eslint` (for `eslint`) or `yarn prettier` (for prettier; running `yarn lint` will fail (due to eslint issues), so it won't display the prettier issues).

## Known issues

* Jest tests of `.tsx` files doesn't work right now.  
  There aren't any component tests yet, so it's something we can address after merging this PR. But we should address is!
* This repo now uses prettier and eslint directly, which means we can't have the convenience of `d2-style` anymore.  
  But this is in line with what we want to move towards anyway, see [this](https://dhis2.slack.com/archives/CJF3MDK8Q/p1717598809885159?thread_ts=1716899085.320159&cid=CJF3MDK8Q) slack message.
* We can't use `"type": "module"` in the `package.json` (see [here](https://github.com/microsoft/playwright/issues/23662))

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  root: true,
+  env: { browser: true, es2020: true },
```
Comment (HendrikThePendric) [inline .eslintrc.cjs:3]: 👍 we need this to be ES2020 to support optional chaining.

File: .eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  extends: [
+    'eslint:recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:react-hooks/recommended',
```
Comment (HendrikThePendric) [inline .eslintrc.cjs:7]: Perhaps this is the same as before.... I just wonder if we should not use the DHIS2 ESLint rules here?

**Review by HendrikThePendric — APPROVED** [review]
Left a few comments, but only minor points, so LGTM 🚀

**Review by Chisomchima — APPROVED** [review]
lgtm

**Review by tonsV2 — CHANGES_REQUESTED** [review]

File: .prettierrc.cjs [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// const { config } = require('@dhis2/cli-style')
```
[the remark this answers is not in the document]
Comment (Mohammer5) [reply .prettierrc.cjs:1]: Good catch!

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM node:19-alpine AS builder
+FROM node:20 AS builder
```
[the remark this answers is not in the document]
Comment (Mohammer5) [reply Dockerfile:1]: Yes:

* Node 19 is not LTS. In the node-verse [only even version numbers will have LTS](https://nodejs.org/en/about/previous-releases)
* Many libraries support either node 18 or node 20. It's an issue I encountered after upgrading to vite.

File: Makefile [hunk 5]
```diff
@@ -24,7 +24,7 @@ e2e-test-ui:
 	npx playwright test --ui
 
 test:
-	docker compose run test
+	yarn install --frozen-lockfile && yarn test
```
[the remark this answers is not in the document]
Comment (Mohammer5) [reply Makefile:27]: Tests failed, I'm not sure exactly why.
But why do we need docker to run unit tests? Seems a little excessive

File: src/views/databases/upload-database-modal.tsx [hunk 6]
```diff
@@ -1,4 +1,17 @@
-import { Button, ButtonStrip, FileInput, InputField, LinearLoader, Modal, ModalActions, ModalContent, ModalTitle, SingleSelectField, SingleSelectOption } from '@dhis2/ui'
+import {
```
[the remark this answers is not in the document]
Comment (Mohammer5) [reply src/views/databases/upload-database-modal.tsx:1]: prettier was set to 80 chars per line

File: .prettierignore
[the remark this answers is not in the document]
Comment (Mohammer5) [reply .prettierignore:1]: I wouldn't, just what causes issues. I'd prefer if everything is aligned with our code-style

File: Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (tonsV2) [reply Dockerfile:1]: But we can still use node:20-alpine, right?

File: Makefile (hunk 5 above)
[the remark this answers is not in the document]
Comment (tonsV2) [reply Makefile:27]: The philosophy is that you only have one dependency (which is Docker). You don't have to deal with versions etc.

File: src/views/databases/upload-database-modal.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (tonsV2) [reply src/views/databases/upload-database-modal.tsx:1]: But we change that back now so this shouldn't be changed, right?

[the remark this answers is not in the document]
Comment (Mohammer5) [reply src/views/databases/upload-database-modal.tsx:1]: I'll address this once the general changes have been approved.
Then I'll run the linter as that will change much more.
I'll request another review from you at that point

**Review by tonsV2 — APPROVED** [review]

**Review by tonsV2 — APPROVED** [review]

**Pull request merged**